core/src/commonMain/kotlin/State.kt

```diff
@@ -104,7 +104,42 @@ public sealed class State {
             public object EncryptionTimedOut : Status()
 
             /** Catch-all for any statuses that are unknown for a platform. */
-            public data class Unknown(val status: Int) : Status()
+            public data class Unknown(val status: Int) : Status() {
+                override fun toString(): String {
+                    return status.toString()
+                }
+            }
+
+            override fun toString(): String {
+                return when (this) {
+                    PeripheralDisconnected -> "Peripheral Disconnected"
+                    CentralDisconnected -> "Central Disconnected"
+                    Failed -> "Failed"
+                    L2CapFailure -> "L2Cap Failure"
+                    Timeout -> "Timeout"
+                    LinkManagerProtocolTimeout -> "LinkManager Protocol Timeout"
+                    UnknownDevice -> "Unknown Device"
+                    Cancelled -> "Cancelled"
+                    ConnectionLimitReached -> "Connection Limit Reached"
+                    EncryptionTimedOut -> "Encryption Timed Out"
+                    is Unknown -> "" // check override
+                }
+            }
```

Comment:
Slightly more condensed as expression body syntax:

```suggestion
            override fun toString(): String = when (this) {
                PeripheralDisconnected -> "Peripheral Disconnected"
                CentralDisconnected -> "Central Disconnected"
                Failed -> "Failed"
                L2CapFailure -> "L2Cap Failure"
                Timeout -> "Timeout"
                LinkManagerProtocolTimeout -> "LinkManager Protocol Timeout"
                UnknownDevice -> "Unknown Device"
                Cancelled -> "Cancelled"
                ConnectionLimitReached -> "Connection Limit Reached"
                EncryptionTimedOut -> "Encryption Timed Out"
                is Unknown -> "" // check override
            }
```